Engine/csound_orc_compile.c

```diff
@@ -1935,10 +1935,13 @@ if (engineState != &csound->engineState) {
       continue;
     if (PARSER_DEBUG)
       csound->DebugMsg(csound, "Instr 0 check on opcode=%s\n", bp->t.opcod);
-    if (UNLIKELY((thread = oentry->thread) & 06 ||
-                 (!thread && bp->t.pftype != 'b'))) {
-      csound->DebugMsg(csound, "***opcode=%s thread=%d pftype=%c\n",
-                       bp->t.opcod, thread, bp->t.pftype);
+    /* VL: now the check is simply for oentry->perf, which is the
+       only condition possible for perf-time code 
+    */
+    if (UNLIKELY(oentry->perf != NULL && bp->t.pftype != 'b')) {
```

Comment:
Rereading the original code, it looks like it checked if it was perf, or thread 0 and bp->t.pftype != 'b'. Guessing this should be:

```
if (UNLIKELY(oentry->perf  != NULL || (oentry->init == NULL && bp->t.pftype != 'b')) {
```

Reply:
going with your solution